### ISSUE TITLE
[2.5D Map] Show direction arrows similar like in the 3D map

### ISSUE THREAD

**Issue by wolfgang-ch** [mytourbook/mytourbook#516]
The current direction arrows are using bitmaps which has UI limits when displayed rotated.

![image](https://user-images.githubusercontent.com/1283445/164023035-1f040f43-4000-4753-977f-78e7994023df.png)

This very first approach is using a circle instead of an arrow 2.5D/3D symbol, this will be the next step but first learning of OpenGL skills is needed

![image](https://user-images.githubusercontent.com/1283445/164023960-ab32dd3e-50c7-4480-bf59-6fcc7b63ab1d.png)

![image](https://user-images.githubusercontent.com/1283445/164957861-cc758221-6a2e-424b-b299-4e389a8b15be.png)

**Comment by wolfgang-ch** [comment]
After many many hours and days of try and error, I could finally setup a dev environment to easily edit/test the vtm sources

**MT commit**
[2.5 Map] Using "vtm-parent" plugin for vtm development https://github.com/mytourbook/mytourbook/commit/7c01221bf98cc860b263239f4f8089f49b8e1adb

**VTM commit**
[2.5D Map] "vtm-parent" plugin setup for vtm development https://github.com/mytourbook/vtm/commit/db276c489586bbd0b495137707901eaf66f9620c

**Comment by wolfgang-ch** [comment]
Finally, now static symbols are painted on the tour path but the more complex math part for me is coming, that the direction arrows are showing in the correct direction

https://user-images.githubusercontent.com/1283445/184494653-259e00e0-de86-42f6-aca4-22462e1ba8d2.mp4

**Comment by wolfgang-ch** [comment]
Direction arrows are now working :-))))

What helped me a lot for the math, examples in the internet, e.g.

* https://stackoverflow.com/questions/31462791/find-coordinates-to-draw-arrow-head-isoscele-triangle-at-the-end-of-a-line#answer-31469874
* https://stackoverflow.com/questions/51369452/get-coordinates-of-arrow-of-a-line-point
* https://math.stackexchange.com/questions/963331/finding-coordinates-of-cap-of-an-arrow

https://user-images.githubusercontent.com/1283445/184596881-d65b6cca-f2a3-45aa-ac03-a3754c51e9ad.mp4

But there are some issues

* Arrows have not yet the same size (easy to fix)
* Some very big arrows, black areas (propably not very easy to fix)
![image](https://user-images.githubusercontent.com/1283445/184597084-ba10d62a-ee17-4221-b0f9-a3f632e56727.png)

**Comment by wolfgang-ch** [comment]
Implementing and debugging 2.5D map features is sometimes creating funny effects

https://user-images.githubusercontent.com/1283445/184815374-0e94a417-8b66-4252-b953-3d573c6c2bc6.mp4

**Comment by wolfgang-ch** [comment]
Should this be the next direction arrow ?

https://user-images.githubusercontent.com/1283445/186596069-c7ceaff8-56a1-4c98-82c8-81509b837e37.mp4

**Comment by telemaxx** [comment]
looks like a Paperplane, nice.

the current position could be a animated bicycle or an walking man. instead of the point.

https://m.youtube.com/watch?v=f3Cr8Yx3GGA

**Comment by FJBDev** [comment]
> looks like a Paperplane, nice.
> 
> the current position could be an animated bicycle or an walking man. instead of the point.
> 
And the MT user will choose their favorite icon in the MT preferences 😄

**Comment by wolfgang-ch** [comment]
> > looks like a Paperplane, nice.
> > the current position could be an animated bicycle or an walking man. instead of the point.
> 
> And the MT user will choose their favorite icon in the MT preferences 😄

... or can add their blender file

**Comment by FJBDev** [comment]
> ... or can add their blender file

That seems advanced for a "normal" user

**Comment by wolfgang-ch** [comment]
Paperplane arrows are now working :-) 

Fin height = 0

https://user-images.githubusercontent.com/1283445/187062914-2a2fecb5-781a-4afd-9513-f8abd1be059f.mp4

Fin height > 0

https://user-images.githubusercontent.com/1283445/187062897-9f1405a9-096e-417a-a80e-1fd85b6adedc.mp4

Currently the fins have a typical 3D issue, z-fighting, the marked fins should not be visible, they are from the arrows which are behind the others.

![image](https://user-images.githubusercontent.com/1283445/187062942-7c3ec440-2854-43e9-b036-2621e6c97cd8.png)

**Comment by wolfgang-ch** [comment]
Z-Fighting is now fixed

![image](https://user-images.githubusercontent.com/1283445/187065075-75501c07-f998-4cf7-a33a-d125b2679391.png)

@telemaxx Can the "old" direction arrows now be replaced with the new arrows ?

**Comment by wolfgang-ch** [comment]
With one middle fin it looks like an area of sharks

![image](https://user-images.githubusercontent.com/1283445/187135721-f4ae6f52-aeef-497c-96ac-a6163bc700a7.png)

![image](https://user-images.githubusercontent.com/1283445/187136144-81e25cf7-bd5a-49af-b1be-dcca4b6a5d05.png)

**Comment by telemaxx** [comment]
> 
> @telemaxx Can the "old" direction arrows now be replaced with the new arrows ?

Yes of course, you're arrows/sharks are looking a lot cooler than my arrows
and they are also visible when the map i tilted a lot.

**Comment by wolfgang-ch** [comment]
The sharks are also in the water

![image](https://user-images.githubusercontent.com/1283445/187214716-4b0d4402-1346-411f-814c-a0f085ed0990.png)

**Comment by wolfgang-ch** [comment]
The arrow head is set to be +1 higher than the arrow tail, when both are on the same z-axis and overlap, then the overlapping part is flickering.

![image](https://user-images.githubusercontent.com/1283445/187867996-fabe5118-fa36-44a0-9807-730aa7abc1ac.png)

**Comment by wolfgang-ch** [comment]
The direction arrow settings are now saved in the track option profiles

https://user-images.githubusercontent.com/1283445/187932207-605e5a8f-f82f-4f68-a3a3-5711153a0b4d.mp4

**Comment by wolfgang-ch** [comment]
There are a few track color issues when the opengl depth buffer is activated for the track. It seems that the line shader algorithm is not supporting track opacity:

* Track corners are flickering when navigating the map
* Track inner color is opaque for the direction arrow wings but the background is not opaque
* Direction arrow wings are painted above the track outline color but should be below


https://user-images.githubusercontent.com/1283445/188626899-518b908a-38ed-4ab3-b628-5e305bcd2fe3.mp4

**Comment by wolfgang-ch** [comment]
Many direction arrows

![image](https://user-images.githubusercontent.com/1283445/189147553-0635759a-292b-4c40-b2de-db749902cf86.png)

![image](https://user-images.githubusercontent.com/1283445/189147578-a1fec987-8f2d-4703-8232-d9df84bf50d1.png)

**Comment by telemaxx** [comment]
good 3D Benchmark.

how was the Performance, could you turn and move in Realtime?

**Comment by wolfgang-ch** [comment]
I recorded this video with 30 fps and it is smoothly


https://user-images.githubusercontent.com/1283445/189275473-97c9402a-19ea-47be-add8-39da8ae5ef2a.mp4


Last week I saw this video where each item is also moved and is not static as the direction arrows

https://www.youtube.com/watch?v=kuy17LVDESk&list=PLRL3Z3lpLmH1EF4g__4gPevJhTArzy0k5

![image](https://user-images.githubusercontent.com/1283445/189275719-348cd914-728d-417b-aec8-500e2f0d80ce.png)

**Comment by wolfgang-ch** [comment]
On the road to a new feature there are sometimes interresting effects.

https://user-images.githubusercontent.com/1283445/190184299-31b2f3c0-da7f-42c4-910b-c0c73ae10661.mp4

**Comment by wolfgang-ch** [comment]
The animation speed can now be selected

https://user-images.githubusercontent.com/1283445/190844735-7655229d-67ae-4897-9e71-727e4e8fae4f.mp4

When the 2.5D map is losing the focus, then the framerate will drop to 1 fps otherwise the selected fps is used. Dropping the fps to 1 will reduce the power consumption (and the pc vent could also stop earlier)

Power consumption with 30 fps

![image](https://user-images.githubusercontent.com/1283445/190846940-288f47fb-df73-4df8-8555-884bb510b932.png)


Power consumption with 1 fps

![image](https://user-images.githubusercontent.com/1283445/190846948-c64fd898-3a10-4b92-83a7-0872ddb40f9e.png)


Is there a need to run animations when the 2.5D map do not have the focus ?

**Comment by wolfgang-ch** [comment]
With https://synthclipse.sourceforge.net/ the shader file editor (c++ editor) has similar features as when it would be a java file, HOWEVER this tool is not compatible with Mylyn tasklist, both need the same plugin with different versions, so I'm using a separate Eclipse instance

![image](https://user-images.githubusercontent.com/1283445/193197627-2c58dd06-a6a9-4b98-8fb4-2c7639762734.png)

**Comment by wolfgang-ch** [comment]
When profiling the OpenGL drawing commands then the directions arrows are consuming the least performance, most is consumed by painting the map

![image](https://user-images.githubusercontent.com/1283445/193313366-9efa7a51-bb0b-45f9-a408-1976b9f59ae5.png)

**Comment by wolfgang-ch** [comment]
This animated triangle is now using a different approach than the prevous animated arrows. The triangle position is now computed in the shader (GPU) and not in the CPU. This allows now to move different models on the tour track.

It took me some day's to finally get a working solution for the first step 😄 but there are additional steps, correct direction, smooth moving, ... 👷 

https://user-images.githubusercontent.com/1283445/194008270-808f2274-1b82-429a-a8c2-b3c5cb05f0da.mp4

**Comment by FJBDev** [comment]
Nice :+1:

**Comment by wolfgang-ch** [comment]
One small step further


https://user-images.githubusercontent.com/1283445/194483909-1311c38b-1e26-445d-be3c-9c108ecb9854.mp4

**Comment by wolfgang-ch** [comment]
Now it looks better, OpenGL has invinite possibilities, when you understand it


https://user-images.githubusercontent.com/1283445/194580289-3286428f-d401-4305-90c4-6bc1c2f2b05f.mp4

**Comment by FJBDev** [comment]
> it looks better, OpenGL has invinite possibilities

Does that mean that in the future, MT will require a minimum graphic card specification (CPU, RAM) to use the 2.5 D map ?

**Comment by wolfgang-ch** [comment]
> > it looks better, OpenGL has invinite possibilities
> 
> Does that mean that in the future, MT will require a minimum graphic card specification (CPU, RAM) to use the 2.5 D map ?

The 2.5D map is mostly not settting a GLSL version, so they use version 110. I just saw that version 120 is set for the circle drawer.

For the direction arrow shaders only, I raised the version to 330 to can use the "modern" (12 year old) in/out syntax.

Until now I don't know which version I need in the future but a new feature could be set optional that the shader do not need to be loaded, which is currently not yet implemented, currently all shaders are loaded at once.

https://en.wikipedia.org/wiki/OpenGL_Shading_Language

![image](https://user-images.githubusercontent.com/1283445/194689585-782b171e-21ce-4eba-bc48-ba80b2ef0cee.png)

**Comment by wolfgang-ch** [comment]
The smoothed animation is now working 😄 

The minimal angle which is set in the video to 3°, could be set to be even more smooth.

https://user-images.githubusercontent.com/1283445/195176293-4dc232cd-be9d-4cd3-8a07-662ee1bdc2f5.mp4

**Comment by wolfgang-ch** [comment]
I started to implement a map player that the animation can be controlled but there are still many desired features which needs to be implemented, e.g. start/stop/pause

https://user-images.githubusercontent.com/1283445/196047207-e9dc985b-b34d-46c9-82f2-dd3a9b991fdc.mp4

**Comment by FJBDev** [comment]
> I started to implement a map player that the animation can be controlled but there are still many desired features which needs to be implemented, e.g. start/stop/pause

Very nice feature!! 👍 
Can you implement it in a way so that the player can be reused for the 2D and 2D maps ?

**Comment by wolfgang-ch** [comment]
It's already prepared for any map but animations in 2D map is total different than in 2.5D or 3D map

![image](https://user-images.githubusercontent.com/1283445/196094400-09de27b7-4f0f-4d44-b1ba-f30e735c1c46.png)

**Comment by wolfgang-ch** [comment]
The map player is now implemented https://github.com/mytourbook/mytourbook/issues/948

**Comment by wolfgang-ch** [comment]
Moved the computation of the animated angle into the shader preparation code, the angle is now computed in each frame and not computed for all frames before the animation starts.

The reason was, when the animation is looped, then the symbol jumped when moving from the end to the start but I didn't thought, that it's now also smoothed when the animation is paused, it is turning the symbol into the correct direction 😄 

https://user-images.githubusercontent.com/1283445/196899914-a880949a-d967-4967-ad28-2950dceb78e0.mp4

https://user-images.githubusercontent.com/1283445/196899908-f9d3a86e-8148-40d9-a9c1-616846d53bf1.mp4

**Comment by FJBDev** [comment]
Will [that "relive" feature](https://github.com/mytourbook/mytourbook/discussions/914#discussioncomment-3654668) be in 22.next ?

**Comment by wolfgang-ch** [comment]
Now with re-live https://github.com/mytourbook/mytourbook/commit/816164349000c8888aa81e4fbb57ab159059079f

https://user-images.githubusercontent.com/1283445/197123523-88fffd1d-2c69-4557-8078-70d1dd9fa498.mp4

**Comment by wolfgang-ch** [comment]
I've found now a promising glTF loader and viewer https://github.com/javagl/JglTF/issues/79#issuecomment-1290052404 which I'll try to use in MT to replace the moving triangle

**Comment by wolfgang-ch** [comment]
Now I'll try to implement the animated figure with the [jMonkeyEngine](https://jmonkeyengine.org/) and then draw (blend) this animation image (Texture2D) into the 2.5D map, per frame.

All of this is done with OpenGL but yesterday I discovered that there are now 2 programs (contexts) using the same OpenGL environment at the same time. This is however supported since a long time and [objects can be shared between different OpenGL contexts](https://www.google.com/search?q=opengl+share+texture+between+contexts ) but it makes the whole idea even more complex. 

What me helps is that there are examples in the web, e.g. https://gregnott.wordpress.com/some-usefull-facts-about-multipul-opengl-contexts/

This is the idea for my next MT steps. Does it work, theoretically yes but the time will tell.

**Comment by wolfgang-ch** [comment]
jMonkeyEngine seems to be not a simple solution https://hub.jmonkeyengine.org/t/use-jmonkeyengine-with-an-external-main-loop/46164 now plan B, I'll try to use libGDX but the current used version must be lifted up in the VTM map otherwise the gltf models do not work https://github.com/mapsforge/vtm/discussions/971

**Comment by wolfgang-ch** [comment]
Now I try to use https://github.com/mgsx-dev/gdx-gltf for libGDX animations and found some nice free models on https://sketchfab.com/ like this bicycle.

However the way is still very long to replace the above animated triangle with an animated 3D model but now it seems that I've found everything which is needed to implement it.

https://user-images.githubusercontent.com/1283445/202741924-def9500c-a732-4b9f-8f40-8fd473a3ff80.mp4

**Comment by FJBDev** [comment]
That's fancy 3D 😄  Can't wait to see what it looks like in MT!

**Comment by wolfgang-ch** [comment]
Today I discovered that there is a 3D poi layer https://github.com/mytourbook/vtm/blob/332a694d653db2d49e044f3b91945e014766319a/vtm-gdx-poi3d/src/org/oscim/gdx/poi3d/Poi3DLayer.java#L63 for the 2.5D map 😄 

However it gets very slow when increasing the window size but it is using libGDX models which I also will use for the map animation, so it seems to be a good example where I can learn from.


https://user-images.githubusercontent.com/1283445/203366311-59ab8b80-b438-47a3-8bdc-cc7d1c4b8996.mp4

**Comment by telemaxx** [comment]
sometimes it can be easy.

did the models produce a shadow like the buildings?

**Comment by wolfgang-ch** [comment]
> sometimes it can be easy.

It looks now easy but it's not. Now I've 2 different running implementations which I have to merge somehow. Both are using the same `com.badlogic.gdx.graphics.g3d.Model` but the environment is different, however this finding is still very helpfull.

> did the models produce a shadow like the buildings?

Currently I don't care about details 😄 

When I get this animation to be working, then shadows could be implemented with libGDX and other cool things.

**Comment by wolfgang-ch** [comment]
The glTF loader has also a demo program where you can test some configurations, also the shadow


https://user-images.githubusercontent.com/1283445/203706989-a30e86f9-4c38-4c2a-9f74-c22d2e030adb.mp4

**Comment by wolfgang-ch** [comment]
This is the very very first running 3D animation with the 2.5D map 😎 

OK, there are issues with the car but a rotation should fix it 😄 

Yesterday evening the code almost worked but it could not find the path to the shader code and I was too tired to get it fixed. Today I added the jar to the vtm plugin and then it worked 😄 

https://user-images.githubusercontent.com/1283445/203915439-d8e89ced-d446-4a67-be08-64e1639e4f8a.mp4

**Comment by wolfgang-ch** [comment]
The animation is not yet perfect, it could be more smooth and better positioned at the head of the triangle, but it works now along a tour track

https://user-images.githubusercontent.com/1283445/204142511-ec7aabda-ea9e-4c7e-9b27-bf0db7f9ed08.mp4

**Comment by wolfgang-ch** [comment]
With my first blender experience I could change the color/texture of the skateboard to be more colorful

From 

![image](https://user-images.githubusercontent.com/1283445/204994307-7e35de4c-9900-4ffe-b32f-97e5345bc201.png)


To


https://user-images.githubusercontent.com/1283445/204994415-afc8dc9c-bdd3-4493-8846-bd134610d2df.mp4

**Comment by wolfgang-ch** [comment]
The model is now moving very smoothly even when this video do not look always smoothly. 

The map location is now also automatically adjusted to the model position which is centered in the map after the animation has ended.

There are still issues but it's now better than before 😄 


https://user-images.githubusercontent.com/1283445/211164099-094fd66e-1675-4f2d-b1f7-e86411d7f17c.mp4

**Comment by wolfgang-ch** [comment]
The autoplayed model can now be centered in the map view.

https://user-images.githubusercontent.com/1283445/212932829-a30a38d8-ce21-4d71-ba23-3607839aa21d.mp4

**Comment by wolfgang-ch** [comment]
The plane is now turning better

https://user-images.githubusercontent.com/1283445/213378004-72e28aae-74c7-4577-be0e-e64defd63dcd.mp4

**Comment by wolfgang-ch** [comment]
With photos is looks more realistic however it looks jerky in the video but it was not on the screen. I recorded it with my `NVIDIA GeForce RTX 2070 SUPER` graphics card.


https://user-images.githubusercontent.com/1283445/213492273-dfe6abe4-7b42-4440-b30b-b0a73ffa7759.mp4


Here are the protagonists from this flight, pilot Martin and me

![plane-1000](https://user-images.githubusercontent.com/1283445/213492585-bb88e331-155c-4b48-8d04-7f0a9d1b5889.jpeg)

**Comment by FJBDev** [comment]
The more I follow what you are developing here, the more it makes me think of a feature I always had in the back of my mind:
- Be able to take 2 identical tours (or more ?) and replay them live to  compare them visually. Below is what my vision is of it:
Example: When I run or bike a route that I do often, I'd like to compare of where I lost time and where I gained time for each tour

Note in highlighted red, the instant comparison of which tour is ahead

![image](https://user-images.githubusercontent.com/25235665/213587468-a92a6ec4-50fd-4057-9068-a4817b3b034b.png)

**Comment by wolfgang-ch** [comment]
I've never thought, that the 2.5D map should be displayed multiple times which is the reason why this map can be currently displayed only once and contains many static fields for just one instance. OpenGL can have multiple views but this would be time consuming to implement it like any other OpenGL code.

An alternative (but not so cool) could be to compare tours within the tour chart, they can be displayed multiple times for different tours, you just need a player for the tour chart similar like the new map player. 

One of my next steps is that the model movement speed is according to the distance or time, currently it is according to the optimized track which skips time slices, except when you zoomed in very very close.

**Comment by wolfgang-ch** [comment]
I'm also using this track to test the moving model and here you can see how the speed is currently implemented and it's problems: 

When a distance between 2 points is larger then the speed is higher and vice versa, on a "normal" track this is not so good visible that you can recognize it.

https://user-images.githubusercontent.com/1283445/213640612-f53244fa-7a54-469b-b88d-e544c4a34583.mp4

**Comment by FJBDev** [comment]
> I've never thought, that the 2.5D map should be displayed multiple times which is the reason why this map can be currently displayed only once and contains many static fields for just one instance. OpenGL can have multiple views but this would be time consuming to implement it like any other OpenGL code.

Maybe then having multiple objects on a single 2.5D map ? A 2D map would be fine too

> 
> An alternative (but not so cool) could be to compare tours within the tour chart, they can be displayed multiple times for different tours, you just need a player for the tour chart similar like the new map player.

Like you say, it's not so cool because a map, I think, is more easy for a user to see what the current location is than on a graph/chart. But having both the chart (bottom of my screenshot) and the map (top of my screenshot) would help

**Comment by wolfgang-ch** [comment]
When a model is imported, then mostly it is not looking in the forward direction and the head is not at the correct position, this can be adjusted in the import dialog. With the Live update feature it can be easily adjusted.


https://user-images.githubusercontent.com/1283445/216286882-ee80c1d7-4a7f-4482-99ad-d5c2705c1c2a.mp4

**Comment by wolfgang-ch** [comment]
The animated model is now merged into the main branch and can be tested 😃 

Here are some hints and known issues

## Start Animation
Currently the animation can be started only this way in the `Track Options` slideout
* Show direction arrows
* Animate

Changing this procedure is on my todo list at the top

![m25-start-animation](https://user-images.githubusercontent.com/1283445/216807320-e1bd9703-d0a0-4df9-8115-0d7145776154.png)



## Center Model

The model can be centered in the map when the map sync action is selected

![m25-center-model](https://user-images.githubusercontent.com/1283445/216807717-acbc7e1e-7ba4-43cd-a230-e66762fa0fff.png)



## Hidden Model

During loading of the map tiles, the model can be flickering or is partly hidden

![m25-invisible](https://user-images.githubusercontent.com/1283445/216807553-38c565ef-cd04-4ace-a8a8-0f71770cff4f.png)



## Model Issues

* The model head position cannot be positioned correctly in all cases, in this case the model is moved to the side and not forward/backward
* More complex models can have textures, transparency or color issues. The high wheeler looks good but not the wood plane/truck

![m25-head-position](https://user-images.githubusercontent.com/1283445/216807692-3a03c117-ca30-44d8-bee5-d66e7d609591.png)



## Re-live 
Is not working properly

![m25-re-live](https://user-images.githubusercontent.com/1283445/216807478-68ffef84-f909-4694-a3f8-c6d3bdeeaa6b.png)



## Background FPS

Is not working properly when enabled

![m25-background-fps](https://user-images.githubusercontent.com/1283445/216807816-5cd23552-8997-470c-be65-d27b218a12bb.png)

**Comment by telemaxx** [comment]
Breaking News: Monster Skateboard in Hanover!

![grafik](https://user-images.githubusercontent.com/22706487/216820266-2077f5c8-069f-4055-a027-4b1650ff20c0.png)

**Comment by wolfgang-ch** [comment]
> Monster Skateboard in Hanover!

This morning I increased the model max size from 1'000 to 10'000 to be able to can see more details

**Comment by wolfgang-ch** [comment]
This is a monster skateboard with the model size of 2000, so there is room to the max.

That the model is better visible, the map tilt rotation axis can be adjusted which I did for this picture

![m25-monster-skateboard](https://user-images.githubusercontent.com/1283445/216939844-6bd090c8-d945-4a9e-b169-dbdee4e12493.png)


There is a nice effect when the map model and direction arrows are displayed at the same time because the skateboard is slightly bumping up and down.

https://user-images.githubusercontent.com/1283445/216940605-93e009a8-54a4-467f-94bd-76937b520ead.mp4

**Comment by wolfgang-ch** [comment]
I'm using the [FastStone capture tool](https://www.faststone.org/FSCaptureDetail.htm) to record the videos. Until now I had to convert the recorded video from WMV format into MP4 by using an external tool, I used https://www.freeconvert.com/wmv-to-mp4

Since a few days, this capture tool can now also save directly into the MP4 format 😎  
https://www.faststone.org/FSCaptureDetail.htm#History

**Comment by FJBDev** [comment]
I've used several times the open-source project [OBS Studio](https://obsproject.com/) and i had a good experience with it

**Comment by wolfgang-ch** [comment]
The model + model cursor can now be displayed/hidden in the tour player view

![image](https://user-images.githubusercontent.com/1283445/217553334-a6c4a45c-034c-4ff7-804a-0cade59112f7.png)


The model turning is now more realistic, the higher the turning angle the more it turns, before it was a static value. It was not so complicated to implement it, than what I first thought.

https://github.com/mytourbook/mytourbook/blob/ed2ded37846cce65be4d25f7f94e28aa4cbd14c2/bundles/net.tourbook/src/net/tourbook/map/player/MapPlayerManager.java#L994-L1003

https://user-images.githubusercontent.com/1283445/217553750-260a17b7-a56b-4c75-9200-8a8d8205b588.mp4

**Comment by wolfgang-ch** [comment]
re-Live is now partly fixed

* Flickering when zoomed in/out
* Track painting is almost correct

The painted track is almost correct when just one tour is displayed but with many tours are displayed then the painted track is more and more wrong

What will not be fixed is the jerky painting of the track this would need a whole new track painting algorithm -> too complicated


https://user-images.githubusercontent.com/1283445/218138293-289543c9-7668-44b6-95c6-486fbf55aeea.mp4

**Comment by wolfgang-ch** [comment]
re-Live stress test with all my tours


https://user-images.githubusercontent.com/1283445/218503421-db5d5c06-2309-4ea2-af78-e8ad893f53e9.mp4

**Comment by wolfgang-ch** [comment]
Finally, this feature is now production ready 😎

**Comment by wolfgang-ch** [comment]
Creating an installable new version fails currently because the glTF model files are in the product files but are not expanded/saved where they can be loaded

![image](https://user-images.githubusercontent.com/1283445/220944565-ba408a46-40a0-4c70-a23e-1a75ad175b37.png)

![image](https://user-images.githubusercontent.com/1283445/220943526-4b7a5cf5-c0f5-44eb-810a-f3dd7317bc44.png)
